packages/cli/src/executions/executions.controller.ts

```diff
@@ -30,37 +24,37 @@ export class ExecutionsController {
 			: await this.workflowSharingService.getSharedWorkflowIds(user, ['workflow:owner']);
 	}
 
-	@Get('/')
+	@Get('/', { middlewares: [parseRangeQuery] })
+	@RequireGlobalScope('workflow:list')
 	async getMany(req: ExecutionRequest.GetMany) {
-		const workflowIds = await this.getAccessibleWorkflowIds(req.user);
+		const accessibleWorkflowIds = await this.getAccessibleWorkflowIds(req.user);
 
-		if (workflowIds.length === 0) return { count: 0, estimated: false, results: [] };
+		if (accessibleWorkflowIds.length === 0) {
+			return { count: 0, estimated: false, results: [] };
+		}
 
-		return await this.executionService.findMany(req, workflowIds);
-	}
+		const { rangeQuery: query } = req;
 
-	@Get('/active')
-	async getActive(req: ExecutionRequest.GetManyActive) {
-		const filter = req.query.filter?.length ? jsonParse<GetManyActiveFilter>(req.query.filter) : {};
+		if (query.workflowId && !accessibleWorkflowIds.includes(query.workflowId)) {
+			return { count: 0, estimated: false, results: [] };
+		}
 
-		const workflowIds = await this.getAccessibleWorkflowIds(req.user);
-
-		return this.isQueueMode
-			? await this.activeExecutionService.findManyInQueueMode(filter, workflowIds)
-			: await this.activeExecutionService.findManyInRegularMode(filter, workflowIds);
-	}
+		if (!query.status || query.status?.length === 0) {
+			const [active, latestFinished] = await Promise.all([
+				this.executionService.findAllActive(),
+				this.executionService.findLatestFinished(20),
```

Comment:
@ivov @krynble could this be the problem why the frontend doesn't load more executions?

Reply:
Cannot reproduce, pinged you to sync on this.